app/models/blueprint.rb

```diff
@@ -4,6 +4,7 @@ class Blueprint < ApplicationRecord
 
   virtual_has_one :bundle
   virtual_has_one :content, :class_name => "Hash"
+  virtual_column :in_use?, :type => :boolean
```

Comment:
There are so many virtual relationships in play here.
This looks like the best you can get it.